### ISSUE TITLE
HDR type icon on start of video for showing what type of video is playing

### ISSUE THREAD

**Issue by frodo19** [b-jesch/skin.estuary.modv2#142]
Hi Jesch,
its not an issue, just request.
Please check that, if as a development possible is.
When the movie begin to playing, would be  shown in the fullscreen of the  left corner, the hdr videotype icon, in 6-8 sec the auto fade/dissapear.
When the movie sdr, will be nothing shown.

![fanart10](https://github.com/b-jesch/skin.estuary.modv2/assets/1296125/b72f3d2f-9558-4838-b1b8-9184e22dcc98)
![fanart101](https://github.com/b-jesch/skin.estuary.modv2/assets/1296125/5e4aa542-fe37-42bc-8f37-393214bac336)
![fanart102](https://github.com/b-jesch/skin.estuary.modv2/assets/1296125/81a48a20-23e0-464a-bc11-c70ad9c9a4ea)

thanks

**Comment by C4Wiz** [comment]
That is the job of your display to show the icon of the current video, mine is a LG C3 and displays in the upper right at the start of the video

**Comment by frodo19** [comment]
> That is the job of your display to show the icon of the current video, mine is a LG C3 and displays in the upper right at the start of the video

Some display made, some not. 
Kodi can handle theese formats now, so when user choice, to show, why not? Its an truly improvements of the skin.

**Comment by C4Wiz** [comment]
> > That is the job of your display to show the icon of the current video, mine is a LG C3 and displays in the upper right at the start of the video
> 
> Some display made, some not. Kodi can handle theese formats now, so when user choice, to show, why not? Its an truly improvements of the skin.

If the display supports the formats it will have a icon. They ALL do!!!

**Comment by frodo19** [comment]
> > > That is the job of your display to show the icon of the current video, mine is a LG C3 and displays in the upper right at the start of the video
> > 
> > 
> > Some display made, some not. Kodi can handle theese formats now, so when user choice, to show, why not? Its an truly improvements of the skin.
> 
> If the display supports the formats it will have a icon. They ALL do!!!

THE FIRST as I wrote,  not all display made that.

Second, when the kodi switch the resolution , the black screen sometimes overwrite the display time of the badge.

Third, the request I made for Jesch, not for you.

Thanks for understanding.

**Comment by b-jesch** [comment]
Keep cool ;)

I agree with @C4Wiz, the display of the video type flag says nothing about the capability of displaying video types of the hardware chain of your setup. The flags just tell only about the video type inside the file but nothing about the display capability of the components (Kodi hardware, receiver, tv) within the signal chain. So it's confusing to see a "Dolby Vision" flag on the tv when a tv or receiver is only capable to process HDR/HLG or just SDR files, as this implies that the hardware is capable playing DV correctly.

Some devices (like my Hisense TV) doesn't display the video type at startup of a movie, but it's showing the signal properties within the info screen. This is the only (and correct) indication of a properly working signal chain.

**Comment by frodo19** [comment]
> Keep cool ;)
> 
> I agree with @C4Wiz, the display of the video type flag says nothing about the capability of displaying video types of the hardware chain of your setup. The flags just tell only about the video type inside the file but nothing about the display capability of the components (Kodi hardware, receiver, tv) within the signal chain. So it's confusing to see a "Dolby Vision" flag on the tv when a tv or receiver is only capable to process HDR/HLG or just SDR files, as this implies that the hardware is capable playing DV correctly.
> 
> Some devices (like my Hisense TV) doesn't display the video type at startup of a movie, but it's showing the signal properties within the info screen. This is the only (and correct) indication of a properly working signal chain.

Allright, when not doable I can see.

Mostly use Coreelec whitch not just show the display capability , then indicate the correct output too in playerprocessinfo 
Player.Process(amlogic.eoft_gamut)    DV , HDR or HDR10+
When any hw of the chain not capable, these info will not shown.
I thinked these infos can use, for display, but its ok.
thx

**Comment by b-jesch** [comment]
In my opinion Player.Process(amlogic.eoft_gamut) is only available on Amlogic devices. This needs a special handling. Will take some investigation and further testing...

**Comment by frodo19** [comment]
Yes, thanks for the work, and time on this.
I have some Coreelec dolby vision capable box.
The hdr/dv  type movies output result, and the eotf can see in the picture.

Mabe can use for display these as icon, some similar path ... :  
<param name="visible" value="String.Contains(Player.Process(amlogic.eoft_gamut),DV" />
<param name="visible" value="String.Contains(Player.Process(amlogic.eoft_gamut),HDR10" />
<param name="visible" value="String.Contains(Player.Process(amlogic.eoft_gamut),HLG" />
but for HDR10+ no idea..

![ce_play](https://github.com/b-jesch/skin.estuary.modv2/assets/1296125/115b12cc-871f-4cb0-b2fb-2907c83bc307)

**Comment by b-jesch** [comment]
The pictures above showing the modes from CoreElec (System Process Info) or from your TV? My Box (Tanix TX3) shows always
HDR10 regardless if playing HDR, HDR+ or DV even if capable to display all three formats. Displaying a HDR mode from Android based on EOTF/Gamut is useless in this case.

**Comment by frodo19** [comment]
> The pictures above showing the modes from CoreElec (System Process Info) or from your TV? My Box (Tanix TX3) shows always HDR10 regardless if playing HDR, HDR+ or DV even if capable to display all three formats. Displaying a HDR mode from Android based on EOTF/Gamut is useless in this case.

Yes. its from my Homatics box r 4K plus, run corelelec CE21 ne nightly. My box, and the tv can handle all this format.
Maybe the tanix can just hdr10 format.

**Comment by frodo19** [comment]
Btw. I tryed to use your modifications from github, and all icon working and display correctly, just the hdr10+ show symple hdr10 icon.
Maybe the kodi cannot read that + ?
 Coreelec shows in eotf HDR10+
thanks

**Comment by b-jesch** [comment]
https://forum.kodi.tv/showthread.php?tid=375547

**Comment by frodo19** [comment]
> https://forum.kodi.tv/showthread.php?tid=375547

Thanks, yes, I tryed the String.EndsWith(Player.Process(amlogic.eoft_gamut),HDR10+) but thats not working too. 
Anyway here is the eotf handling in coreelec.
https://github.com/CoreELEC/CoreELEC/blob/a92f6ea60b2f927f917c4053b677d05d82cb7f2e/projects/Amlogic-ce/devices/Amlogic-ng-dv/patches/linux/0016-hdmitx-add-dolby-vision-support-for-display-current-.patch#L19

**Comment by b-jesch** [comment]
Could you try this in Variables.xml?
```
<value condition="String.Contains(Player.Process(amlogic.eoft_gamut),+)">flags/fullscreen/hdrplus_fs.png</value>
```

**Comment by frodo19** [comment]
> Could you try this in Variables.xml?
> 
> ```
> <value condition="String.Contains(Player.Process(amlogic.eoft_gamut),+)">flags/fullscreen/hdrplus_fs.png</value>
> ```
Yep, I tryed, no improvements. Display just the hdr logo. I tryed remove the hdr10 related line , then no logo.

**Comment by b-jesch** [comment]
https://github.com/xbmc/xbmc/issues/24337

**Comment by frodo19** [comment]
Its kodi bug seems. I hope they can do anything with that. 
But thanks for your modifications, I have not lot hdr10plus files.
All other works fine in CoreElec now.

**Comment by Sunset-1982** [comment]
Could you add an option to select a delay or the duration of how long the the Icon is shown? for example 3,5,7 or 10 Seconds at the start of the video. Becaus my tv needs a few seconds for the hdmi handshake and when its done, the dolby vision picture has already disapeard.

**Comment by frodo19** [comment]
> Could you add an option to select a delay or the duration of how long the the Icon is shown? for example 3,5,7 or 10 Seconds at the start of the video. Becaus my tv needs a few seconds for the hdmi handshake and when its done, the dolby vision picture has already disapeard.

It 7 sec now, not enough?  3, and 5 will not usable.. But maybe 7,9,11..

**Comment by Sunset-1982** [comment]
really? I can see it for 1 second max, sometimes I don't see it at all. The hsmi handshake takes a while. A useroption to set it between 5 and 15 seconds would be nice, or some steps like 7,, 10 and 12

**Comment by b-jesch** [comment]
I've implemented 5, 7, 10, 15 secs now.

**Comment by Sunset-1982** [comment]
Wow! Very nice! Thank you for the fast response!

**Comment by frodo19** [comment]
Hi jesch,
I really like this feature in Coreelec.
But the latest CE nightly broke for me.
Until 19 may worked fine.
The HDR sometime shown, the DV not shown anymore.
The HDR icon when playing, and then stop, and playing again sometime dont show again until the skin reboot.
The DV icon not show anymore.
I see the skindebug activate, sometime the Custom_1111_FullScreenResolutionInfo.xml not load at start, just the VideoFullScreen.xml.

The last changes was in Corelelec [here](https://github.com/CoreELEC/xbmc/commit/030de180ea90a34d9419a112eb3b06ed15b023a9#diff-2957a3520c1e3844f8b549ee5d2fc8947bc8575a5b415565855c56b4dd53cb57R67-R76)
When you have time, please see whats broken now.
Thanks you

**Comment by b-jesch** [comment]
Loading of Custom_1111 is conditional: https://github.com/b-jesch/skin.estuary.modv2/blob/3316820ad948e55005839398c5911ecd14ca9720/xml/VideoFullScreen.xml#L6
For the moment there's nothing I can do. I suggest using the last stable version and not nightlies.
